### ISSUE TITLE
Flak Cannon

### ISSUE THREAD

**Issue by HellDragon-HK** [GreatEmerald/UT3Weapons#8]
- GAME PLAY RELATED
  - [ ] Flak shell pickup worth 10 shots in UT2004 and only 8 in UT3
  - [ ] Alt fire may have a higher arc in UT3
  - [ ] UT3 Flak has a consistent middle chunk, UT2004 does not

 - SOUND RELATED
  - [ ] Impact Flesh when hitting players is not used
  - [ ] Although they didn't do it, there are Dirt, Metal, & Stone impact sounds that could be used separately for select surface types instead of a random group like UT3 has them  

 - VISUAL EFFECTS RELATED
  - [ ] UT2004 muzzle flash is small and in the center of the weapon while in UT3 it's the the entire weapon 
  - [ ] I never noticed the UT3 Flak brought back the number indicator from UT99, doesn't look like it works on UT2004 though
 - [ ] Flak primary is more of an orange color in UT3
 - [ ] UT3 Flak primary looks thicker and even spins
 - [ ] UT3 Alt fire shell has orange fire with smoke like a rocket look to it where as the UT2004 one just has smoke
 - [ ] The Alt fire impact explosion is more spectacular in UT3, possibly could be considered similar to an SPMA shell explosion
 - [ ] Impact from primary fire in UT3 leaves a short lived orange circle, in UT2004 it doesn't appear to leave anything

**Comment by GreatEmerald** [comment]
Yea, the LED panel doesn't really seem worth coding it. It could in theory be done though.

Fun fact: you can in fact snipe with the Flak in UT3. The centre shard is always pinpoint accurate. Though I guess in this case you mean the spread, which might need some adjustments, indeed.

**Comment by HellDragon-HK** [comment]
Almost every shot from the flak has long range ability from what I've seen I'll keep a closer eye on it to try and be sure

**Comment by HellDragon-HK** [comment]
More inconsistent damage and wow was this one off
Primary Fire - UT3 100 - 15 UT2004 100 - 46 how's that possible?
Ok standing a little closer to the wall in UT3 I got 100 - 37 which still isn't the same but it's a lot closer

**Comment by GreatEmerald** [comment]
That's because in UT2004 flak chunks may bounce or may not, that's random; plus the damage you take depends on the velocity of each chunk.
It might be worth investigating the difference in the formulæ used to calculate the damage between the games...

**Comment by HellDragon-HK** [comment]
Something I didn't try is that it seems to take 3 flak shells aimed directly at the ground on UT3 to kill yourself with and with UT2004 I think it only takes 1 but I'll need to check, currently the map I'm on doesn't seem to have a flak.

**Comment by GreatEmerald** [comment]
No, a flak chunk does 18 points of damage max if I'm reading the code right.
EDIT: Oh, you mean the shell. Needs looking into, then.

**Comment by HellDragon-HK** [comment]
Yeah I need a map with a flak and I'll check it, right now I'm having issues testing Alex's new textures, either I missed a file or something isn't working

**Comment by HellDragon-HK** [comment]
New UT3Weapons compiled....

The alt fire on the UT3 Flak is going through the floor on the first shot every time when aimed directly at the ground and I did 4 times in a row to be sure

Alt fire damage is about the same, I get around 47 health left on UT2004 after 1st shot and in UT3 it's been 57-40 with the 57 I had to do it 3 times which brought it down to 11 health....some difference in damage maybe but probably not too much

EDIT-Huh bizarre now it's not going through the floor
EDIT2-Yeah now it's not happening, after using the windows key to exit to report the issue and going back in it stopped happening
EDIT3-Oh there I got it again, apparently you can fire it through the cracks on Idoma near it's weapon base and doing in a row like that was just luck of some kind not sure if it's good or bad since it's testing lol

**Comment by HellDragon-HK** [comment]
Flak damage is too inconsistent in both games to check, UT3 damage on vehicles ranged from 86-126 or so (Hellbender went from 600 to 474 and Scorpion from 300 to 176 but not consistently as it was only 98 on first try for Scorpion)

Flak primary does not cause knockback on Scorpion in UT3 but in UT2004 causes little to fair and flip
Flak primary causes fair knockback in UT3 and extreme knockback in UT2004

Flak alt causes no knockback to Scorpion in UT3 but in UT2004 causes fair to medium knockback
Flak alt causes fair knockback to Manta in UT3 but in UT2004 causes extreme knockback

Flak primary is causing a little knockback on the Hellbender and lifting it off 2 wheels, in UT3 it almost lifts off 2 wheels but doesn't and has no knockback
Flak alt is causing a little knockback on the Hellbender, it doesn't cause any in UT3

Flak primary causes little knockback in UT3, in UT2004 on Torlan it sends the Raptor flying off it's spawn pad
Flak alt....holy &#H@....in UT2004 it sends the Raptor completely over the wall BEHIND the power core and up the mountain, it doesn't do anything in UT3

Flak primary causes medium to massive knockback on the Cicada in UT2004 and makes it slide like it's own ice, in UT3 it only does little knockback
Flak alt in UT3 does nothing, in UT2004 you can send the Cicada flying over a wall easily

Flak alt moves SPMA around a little, doesn't in UT3

**Comment by HellDragon-HK** [comment]
@GreatEmerald You know when you said you can snipe with the Flak, you can't on UT2004 because there isn't a consistent middle chunk like there is in UT3, took awhile to confirm but I'm positive of it now.
